src/time_utils.c

```diff
@@ -149,7 +153,10 @@ ts_time_value_from_arg(Datum arg, Oid argtype, Oid timetype, bool need_now_func)
 				 errmsg("invalid time argument type \"%s\"", format_type_be(argtype)),
 				 errhint("Try casting the argument to \"%s\".", format_type_be(timetype))));
 
-	return ts_time_value_to_internal(arg, argtype);
+	if (convert_internal)
+		return ts_time_value_to_internal(arg, argtype);
+
+	return DatumGetInt64(arg);
```

Comment:
Can this be moved by caller? Two bool parameters might be a little confusing, especially when the function comment starts with __Get the internal time value from a pseudo-type function argument__, but then it has a parameter that says __actually don't get the internal time value__ :)

Reply:
@akuzm Umm, I have added a comment before the start of the function about the "internal" option. Ok, removed the "internal" mention in the first line of the comment for this function :-)

Reply:
@akuzm any other comments here?

Reply:
@antekresic @akuzm Ok, I removed the bool flag and added explicit implicit conversions now.